store/tikv/lock_resolver.go

```diff
@@ -457,6 +457,20 @@ func (lr *LockResolver) getTxnStatusFromLock(bo *Backoffer, l *Lock, callerStart
 	for {
 		status, err = lr.getTxnStatus(bo, l.TxnID, l.Primary, callerStartTS, currentTS, rollbackIfNotExist)
 		if err == nil {
+			// If l.LockType is pessimistic lock type:
+			//     - if its primary lock is pessimistic too, the check txn status result should not be cached.
+			//     - if its primary lock is prewrite lock type, the check txn status could be cached.
+			// If l.lockType is prewrite lock type:
+			//     - always cache the check txn status result.
+			// For prewrite locks, their primary keys should ALWAYS be the correct one.
+			if l.LockType != kvrpcpb.Op_PessimisticLock && status.ttl == 0 {
```

Comment:
Seems in the current code, we don't cache when the primary lock is prewrite lock type. Can we move this logic to where `saveResolved` originally was (L579)? We have more information there. And we can even cache it when the primary lock is committed with a positive ts.

Reply:
If we want to cache all results, we need to change kvproto.

> And we can even cache it when the primary lock is committed with a positive ts.
It seldom happens...

Reply:
We could not get the lock type for the lock we met in the original place, which is needed for the cache check.  We could add back the saving logic for commited transaction in the original place?

Reply:
okay i think it's optional as youjiali1995 says this case seldom happens.

Reply:
@youjiali1995 @sticnarf 
In the previous diff the resolved cache would be checked everytime which is not necessary, I put back the cache fill related logic to its original place, PTAL again, thx